Integrate the `alls-green` action into the main GHA CI workflow

x1101 opened the pull request:
Resolves #1284.

Testing shows that this doesn't surfaces which part of  the nox matrix fails, but does fail the workflow when any of them do.

x1101 committed:
Integrating all-greens into ci.yml

webknjaz commented on .github/workflows/ci.yaml:
```diff
@@ -11,3 +11,19 @@ on:
+    - name: Decide whether the needed jobs succeeded or failed
+      uses: re-actors/alls-green@release/v1
+      with:
+        allowed-failures: docs, linters
+        allowed-skips: non-voting-flaky-job
```
There's no such jobs in this workflow.
```suggestion
```

webknjaz reviewed (changes requested):
Almost there! Just drop the non-existing references…

webknjaz commented:
> Testing shows that this doesn't surfaces which part of the nox matrix fails, but does fail the workflow when any of them do.

GitHub does not provide us with this context, so it's not possible to retrieve.

x1101 committed:
removing non-existing references

When rebuilding this work after repo cleanup, I missed removing these. Thanks for catching that.

Co-authored-by: Sviatoslav Sydorenko (Святослав Сидоренко) <wk.cvs.github@sydorenko.org.ua>

x1101 commented:
> > Testing shows that this doesn't surfaces which part of the nox matrix fails, but does fail the workflow when any of them do.
> 
> GitHub does not provide us with this context, so it's not possible to retrieve.

Yeah, I was just highlighting the results of my testing and the limits of what this can provide, because the issue requesting doesn't really specify what information we were trying to get and I wanted it to be obvious what this does/doesn't do

gotmax23 commented:
I'm honestly fine with the current approach. We already require the two docs build/checker jobs in the branch protection settings.

oraNod commented on .github/workflows/ci.yaml:
```diff
@@ -11,3 +11,17 @@ on:
+    needs:
+      - nox
+
+    runs-on: Ubuntu-latest
```
```suggestion
    runs-on: ubuntu-latest
```

webknjaz replied:
@oraNod do you want to commit this from the UI and merge right away?

oraNod replied:
@webknjaz Sure. I can also do it now.

webknjaz commented:
> I'm honestly fine with the current approach. We already require the two docs build/checker jobs in the branch protection settings.

But it requires somebody with proper access to continuously watch for the need to update the branch protection rules whenever workflows change, vs. having a transparent record of what's required.

oraNod committed:
Update .github/workflows/ci.yaml

oraNod commented:
Thanks very much for doing the work here @x1101 and thanks for the review @webknjaz :tada:

samccann commented:
@oraNod can you add any appropriate backport labels (if we want this in anything other than devel for how)?

oraNod commented:
@samccann Shoot. Guess I forgot the backport. Thanks for the reminder. Maybe we should just keep this on devel and stable-2.17.
